vscode-ext/src/tests/e2e.ts

```diff
@@ -0,0 +1,146 @@
+        if (await debugToolbar?.isDisplayed()) {
+            await debugToolbar.stop();
+        }
+    });
+
+    it("enable mirrord", async function () {
+        const statusBar = new StatusBar();
+        await browser.driver.wait(async () => {
+            const enableButton = await statusBar.getItem("Enable mirrord");
+            if (enableButton !== undefined) {
+                enableButton.click();
+                return true;
+            }
+        }, DEFAULT_CONDITION_TIMEOUT, "mirrord `enable` button not found -- timed out");
+
+        await browser.driver.wait(async () => {
+            const enableButton = await statusBar.getItem("Disable mirrord");
+            if (enableButton !== undefined) {
+                return true;
+            }
+        }, DEFAULT_CONDITION_TIMEOUT, "Mirrord not enabled -- timed out");
+    });
+
+    it("create mirrord config", async function () {
+        // gear -> $(gear) clicked to open mirrord config
+        const statusBar = new StatusBar();
+        await browser.driver.wait(async () => {
+            const mirrordSettingsButton = await statusBar.getItem("gear");
+            if (mirrordSettingsButton !== undefined) {
+                mirrordSettingsButton.click();
+                return true;
+            }
+        }, DEFAULT_CONDITION_TIMEOUT, "mirrord config `$(gear)` button not found -- timed out");
+
+        await browser.driver.wait(async () => {
+            return await existsSync(mirrordConfigPath);
+        }, DEFAULT_CONDITION_TIMEOUT, "mirrord `default` config not found");
+    });
+
+    it("select pod from quickpick", async function () {
+        await setBreakPoint(fileName, breakPoint, browser);
+        await startDebugging(configurationFile);
+
+        const inputBox = await InputBox.create();
+        // assertion that podToSelect is not undefined is done in "before" block   
+        await browser.driver.wait(async () => {
+            return await inputBox.isDisplayed();
+        }, DEFAULT_CONDITION_TIMEOUT, "quickPick not found -- timed out");
+        await inputBox.selectQuickPick(podToSelect!);
+    });
+
+    it("wait for breakpoint to be hit", async function () {
+        debugToolbar = await DebugToolbar.create(20000);
+        // waiting for breakpoint and sending traffic to pod are run in parallel
+        // however, traffic is sent after 10 seconds that we are sure the IDE is listening
+        // for breakpoints
+        await browser.driver.wait(async () => {
+            return await debugToolbar.isDisplayed();
+        }, 20000, "debug toolbar not found -- timed out");
+
+        await Promise.all([debugToolbar.waitForBreakPoint(), sendTrafficToPod()]);
+    });
+});
+
+// This promise is run in parallel to the promise waiting for the breakpoint to be hit
+// We wait for 10 seconds to make sure that we are in listening state
+async function sendTrafficToPod() {
+    await new Promise(resolve => setTimeout(resolve, 10000));
```

Comment:
How is this different from a sleep?

Reply:
I cant find a way to make sure that we are listening on the breakpoint, so thought we can sleep for 10 secs and then send the request, wdyt?